app/javascript/mastodon/components/content_warning.tsx

```diff
@@ -0,0 +1,31 @@
+import { FormattedMessage } from 'react-intl';
+
+export const ContentWarning: React.FC<{
+  text: string;
+  expanded?: boolean;
+  onClick?: () => void;
+}> = ({ text, expanded, onClick }) => (
+  <div className='content-warning'>
+    <div className='content-warning__border' />
```

Comment:
Maybe we could use `:before` (and `:after`) here to avoid having to add those to the DOM and have all the styling done in CSS?

Reply:
Would that improve performance?

Reply:
Slightly, having less things rendered in the DOM by React will make it faster (a simpler DOM is always better). It also decouples the document structure from visual appearance, which is always a good idea.